### ISSUE TITLE
perf(color): rgb_to_yuv420 subsamples chroma with a 6-D unfold+mean that is 9x slower than CPU on MPS; avg_pool2d gives the same result 35x faster

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4524]
## Problem

`kornia.color.rgb_to_yuv420` builds the half-resolution chroma planes as

```python
yuvimage[..., 1:3, :, :].unfold(-2, 2, 2).unfold(-2, 2, 2).mean((-1, -2))
```

a mean over two window dims of a 6-D strided view. On MPS that reduction is the whole cost of the op (`aten::mean.dim` 82.8 ms of 84.7 ms at 1024²), and it is 9x slower than the CPU kernel; the op ends up 7.8x slower on MPS than on the M1 CPU while every other colour conversion wins on MPS at that size.

| shape | CPU today | MPS today | CPU with `avg_pool2d` | MPS with `avg_pool2d` |
|---|---:|---:|---:|---:|
| (1,3,1024,1024) | 12.7 ms | **88.8 ms** | 4.6 ms | **2.5 ms** |
| (8,3,256,256) | 6.4 ms | 42.7 ms | – | – |

## Fix

`F.avg_pool2d(yuv[..., 1:3, :, :], 2)` is the same 2x2 box mean: max abs difference against the current output is 3.0e-8 at 1024² (float32). It is also 2.8x faster on the CPU. Same treatment applies to `rgb_to_yuv422` if it uses the same unfold pattern.

```python
import torch, torch.nn.functional as F, kornia.color as KC
img = torch.rand(1, 3, 1024, 1024)
_, uv = KC.rgb_to_yuv420(img)
uv2 = F.avg_pool2d(KC.rgb_to_yuv(img)[:, 1:], 2)
print((uv - uv2).abs().max().item())  # 2.98e-08
```

Precedent for choosing the MPS-friendly op: #3914 / #3916 (`rgb_to_hsv` max/min-with-indices vs amax/amin).

Measured on torch nightly 2.15.0.dev20260913, Apple M1 8 GB, macOS 26.5.1, kornia `633951bd7`, `PYTORCH_ENABLE_MPS_FALLBACK` unset, `torch.utils.benchmark` medians with `mps.synchronize` inside the timed region, fresh process per op. Split out of the MPS-audit tracking issue #4521; the PyTorch-side kernel findings are in #4522.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
